provider.go

```diff
@@ -185,3 +185,31 @@ func (m ProviderModule) postProviderHandler(c *gin.Context) {
 
 	c.JSON(http.StatusCreated, provider)
 }
+
+// putProviderHandler godoc
+// @summary Put provider in database.
+// @description Put provider in database.
```

Comment:
For those uninitiated by REST verbs, maybe it'd be better to change this to something like:

```suggestion
// @description Creates a new provider if a match is not found.
```